### DISCUSSION TITLE
How to share user session between hosts with multiple authelia instances

### DISCUSSION THREAD

**Discussion by Chuckame** [authelia/authelia#6527] [Q&A] [answered]
For homelab and fun purpose, I have 3 hosts: homelab (domotics, dns) & nas in home, and a VPS for seedbox and home tunnel.

Each host can leave without the others, as they each have their own reverse proxy with https and a separate authelia instance with the same credentials everywhere. I usually play with each host, so each of them can be down for any reason.

But I need auth on each host, but to do that, I have to install separate authelia instances, because they don't have a shared database.

I already tried to have the same secrets used by each authelia instances, but it seems not to work (infinite redirect, because local authelia says it's OK, while other authelia request auth). 

Is there anything like jwt (a token signed with private key, and if you have public key, they you can validate that your token is valid) to make stateless auth validation, that allows using the same session cookie on different authelia instances? I've seen the redis sentinel option to sync the db over hosts, but it still need a master...

Note that me and my wife are the only users.

**Comment by james-d-elliott** [comment, marked as answer]
You will need to utilize redis for session storage, this is the only way at this moment of time.
